File: ios/KodexLink/Core/Protocol/Envelope.swift
import Foundation

struct EnvelopeHeader: Decodable {
    let id: String
    let type: String
    let bindingId: String?
    let createdAt: Int
    let requiresAck: Bool
    let protocolVersion: Int
    let idempotencyKey: String?
    let traceId: String?
}

struct Envelope<Payload: Codable>: Codable {
    let id: String
    let type: String
    let bindingId: String?
    let createdAt: Int
    let requiresAck: Bool
    let protocolVersion: Int
    let idempotencyKey: String?
    let traceId: String?
    let payload: Payload
}

File: packages/protocol/src/envelope.ts
import type { ProtocolVersion } from "./versions.js";

export interface MessageEnvelope<TType extends string = string, TPayload = unknown> {
  id: string;
  type: TType;
  bindingId?: string;
  createdAt: number;
  requiresAck: boolean;
  protocolVersion: ProtocolVersion;
  idempotencyKey?: string;
  traceId?: string;
  payload: TPayload;
}

